runtime/sema/check_import_declaration.go

```diff
@@ -45,27 +45,52 @@ func (checker *Checker) VisitImportDeclaration(declaration *ast.ImportDeclaratio
 	})
 }
 
-func (checker *Checker) declareImportDeclaration(declaration *ast.ImportDeclaration) Type {
+func (checker *Checker) declareImportDeclaration(declaration *ast.ImportDeclaration) {
 	locationRange := ast.NewRange(
 		checker.memoryGauge,
 		declaration.LocationPos,
 		// TODO: improve
 		declaration.LocationPos,
 	)
 
-	resolvedLocations, err := checker.resolveLocation(declaration.Identifiers, declaration.Location)
+	identifiers := declaration.Identifiers
+
+	resolvedLocations, err := checker.resolveLocation(identifiers, declaration.Location)
 	if err != nil {
 		checker.report(err)
-		return nil
+		return
+	}
+
+	// 1) If the import is just for an address (e.g. `import 0x01`) without specifying any contracts, AND
+	// 2) If the imported address is same as the address of the account to which the current contract is deploying,
+	// Then this could create a cycle, as soon the current contract is deployed.
+	// e.g:
+	//    import 0x01
+	//    access(all) contract Foo {}
+	//
+
+	if len(identifiers) == 0 {
+		if currentProgramLocation, ok := checker.Location.(common.AddressLocation); ok {
+			if importLocation, ok := declaration.Location.(common.AddressLocation); ok {
+				if currentProgramLocation.Address == importLocation.Address {
```

Comment:
Is there a particular reason why we do this for address locations specifically? Should we generalize this to any location kind?

Reply:
Not sure if we need this for other location types: i.e. AFAIU this can only occur for deployed contracts (hence address locations)